File: ueueo-framework/ueueo-domain-driven-desgin/src/main/java/com/ueueo/ddd/application/contracts/dtos/ExtensibleAuditedEntityDto.cs
using System;
using Volo.Abp.Auditing;
using Volo.Abp.Data;

namespace Volo.Abp.Application.Dtos;

/**
 * This class can be inherited by DTO classes to implement <see cref="IAuditedObject"/> interface.
 * It also implements the <see cref="IHasExtraProperties"/> interface.
*/
 * <typeparam name="TPrimaryKey">Type of primary key</typeparam>
[Serializable]
public abstract class ExtensibleAuditedEntityDto<TPrimaryKey> : ExtensibleCreationAuditedEntityDto<TPrimaryKey>, IAuditedObject
{
     * <inheritdoc />
    public DateTime? LastModificationTime;// { get; set; }

     * <inheritdoc />
    public ID LastModifierId;// { get; set; }

    protected ExtensibleAuditedEntityDto()
        : this(true)
    {

    }

    protected ExtensibleAuditedEntityDto(boolean setDefaultsForExtraProperties)
        : base(setDefaultsForExtraProperties)
    {

    }
}

/**
 * This class can be inherited by DTO classes to implement <see cref="IAuditedObject"/> interface.
 * It also implements the <see cref="IHasExtraProperties"/> interface.
*/
[Serializable]
public abstract class ExtensibleAuditedEntityDto : ExtensibleCreationAuditedEntityDto, IAuditedObject
{
     * <inheritdoc />
    public DateTime? LastModificationTime;// { get; set; }

     * <inheritdoc />
    public ID LastModifierId;// { get; set; }

    protected ExtensibleAuditedEntityDto()
        : this(true)
    {

    }

    protected ExtensibleAuditedEntityDto(boolean setDefaultsForExtraProperties)
        : base(setDefaultsForExtraProperties)
    {

    }
}

File: ueueo-frameworks/ueueo-domain-driven-design/src/main/java/com/ueueo/ddd/application/contracts/dtos/ExtensibleAuditedEntityDto.java
package com.ueueo.ddd.application.contracts.dtos;

import com.ueueo.ID;
import com.ueueo.auditing.IAuditedObject;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.util.Date;

/**
 * This class can be inherited by DTO classes to implement <see cref="IAuditedObject"/> interface.
 * It also implements the <see cref="IHasExtraProperties"/> interface.
 *
 * <typeparam name="TPrimaryKey">Type of primary key</typeparam>
 */
@EqualsAndHashCode(callSuper = true)
@Data
public abstract class ExtensibleAuditedEntityDto extends ExtensibleCreationAuditedEntityDto implements IAuditedObject {

    private Date lastModificationTime;

    private ID lastModifierId;

    protected ExtensibleAuditedEntityDto() {
        this(true);
    }

    protected ExtensibleAuditedEntityDto(boolean setDefaultsForExtraProperties) {
        super(setDefaultsForExtraProperties);
    }
}

